### PR TITLE
Massively speed up the version history update

### PR DESCRIPTION
**Pull request by jskeet**
We don't need to check every file in every commit: if apis/xyz/xyz
has the same sha in the parent of the commit and the "candidate"
commit, then it definitely doesn't contain any changes to API xyz.

This cuts the per-API time down from about 10 seconds to 1 second.

### CODE REVIEW

File: tools/Google.Cloud.Tools.ReleaseManager/UpdateHistoryCommand.cs [hunk 1]
```diff
@@ -17,6 +17,7 @@
 using LibGit2Sharp;
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
```
Comment (chrisdunelm) [inline tools/Google.Cloud.Tools.ReleaseManager/UpdateHistoryCommand.cs:20]: Is this needed?

**Review by chrisdunelm — APPROVED** [review]
LGTM
One questions.

File: tools/Google.Cloud.Tools.ReleaseManager/UpdateHistoryCommand.cs (hunk 1 above)
Comment (jskeet) [reply tools/Google.Cloud.Tools.ReleaseManager/UpdateHistoryCommand.cs]: Nope, sorry - that was left over from using Stopwatch. Removed.

**Comment by jskeet** [comment]
(Will merge on green.)

**Pull request merged**